src/Controls/tests/UITests/Tests/Issues/Issue16561.cs

```diff
@@ -41,19 +45,28 @@ public void TapTwoPlacesQuickly()
 			var expectedX1 = point1.X - tapArea.X;
 			var expectedX2 = point2.X - tapArea.X;
 
-			// Just calling Tap twice will be too slow; we need to queue up the actions so they happen quickly
-			var actionsList = new TouchAction(app2.Driver);
+			if (app2.Driver is WindowsDriver)
+			{
+				// Windows will throw an error if we try to execute Taps with a TouchAction, so we'll just use TapCoordinates instead
+				app2.TapCoordinates(point1.X, point1.Y);
+				app2.TapCoordinates(point2.X, point2.Y);
+			}
+			else
+			{
+				// Just calling Tap twice will be too slow; we need to queue up the actions so they happen quickly
+				var actionsList = new TouchAction(app2.Driver);
 
-			// Tap the first point, then the second point
-			actionsList.Tap(point1.X, point1.Y).Tap(point2.X, point2.Y);
-			app2.Driver.PerformTouchAction(actionsList);
+				// Tap the first point, then the second point
+				actionsList.Tap(point1.X, point1.Y).Tap(point2.X, point2.Y);
+				app2.Driver.PerformTouchAction(actionsList);
```

Comment:
TouchActions are deprecated, why can you not use the abstractions provided by the AppiumUITestApp ?


I don't want to trigger auto tagging so www.github.com/appium/appium/issues/16009

Reply:
> TouchActions are deprecated

I had no idea. No warnings or anything.

> why can you not use the abstractions provided by the AppiumUITestApp

Because calling `TapCoordinates()` twice doesn't work quickly enough to reproduce the original issue on Android.

Reply:
Okay, rewritten for Android using ActionSequence, iOS using ExecuteScript, and Windows using TapCoordinates.